modules/openapi-generator/src/main/resources/Javascript/es6/ApiClient.mustache

```diff
@@ -404,6 +409,23 @@ class ApiClient {
 
         request.query(this.normalizeParams(queryParams));
 
+    {{#detectAmznApiGatewayExtensions}}
+        // Retrieve host from baseURL and full path from baseURL + path
+        var adr = URLS.parse(this.basePath, true);
+        var fullPath = adr.path.substring(1) + path;
```

Comment:
This will throw NPE if we have default route and stage.
You should put a check here unless adr.path always return '/' atleast.
```
var fullPath = path;
if adr.path {
var fullPath = adr.path.substring(1) + path;
} 
```

Reply:
Added.